csrc/mmdeploy/codebase/mmpose/cuda/topdown_affine.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include "mmdeploy/codebase/mmpose/topdown_affine.h"
+
+#include <vector>
+
+#include "ppl/cv/cuda/crop.h"
+#include "ppl/cv/cuda/resize.h"
+
+namespace mmdeploy::mmpose {
+
+class CropAndResizeCudaImpl : public CropAndResize {
+ public:
+  CropAndResizeCudaImpl() = default;
+
+  Result<void> apply(const std::vector<float>& bbox, std::vector<float>& c, std::vector<float>& s,
+                     const std::vector<int>& output_size, const Tensor& src, Tensor& dst) override {
+    OUTCOME_TRY(auto img, MakeAvailableOnDevice(src, device_, stream()));
+    stream().Wait().value();
+
+    // x, y, w, h
+    std::vector<int> roi = {int(bbox[0]), int(bbox[1]), (int)bbox[2], (int)bbox[3]};
+    // update center
+    c[0] = roi[0] + roi[2] / 2.f;
```

Comment:
The `CropAndResize` operation should act as what its name suggests "crop a region (a given rect) and resize it to the given size". The cropping parameters (the region) should not be computed in the operation.